### DISCUSSION TITLE
Expand/Collapse all blocks when editing a page, post or content

### DISCUSSION THREAD

**Discussion by horacioj** [PiranhaCMS/piranha.core#1570] [Ideas] [answered]
In the Manager, there is a very useful configuration setting "Collapse blocks" (_If blocks should be collapsed by default when opening a page or post in the manager._), which works fine and it is great.

My question is, are there "expand/collapse all blocks" buttons in the Manager when editing a page/post/content? I could not find them. If they don't yet exist, I think they could be a nice usability enhancement.

Thanks

**Comment by tidyui** [comment, marked as answer]
Hi there! You could add to your project by creating a file named `_PageExpandCollapse.cshtml` in the folder `~/Areas/Manager/Pages/Partial/Actions/` in your application. The add the following markup to the file:

~~~
<button class="btn btn-primary btn-icon" onclick="expandCollapse(this, false)">
    <i class="fas fa-chevron-down"></i>
</button>
<button class="btn btn-primary btn-icon" onclick="expandCollapse(this, true)">
    <i class="fas fa-chevron-up"></i>
</button>
<script>
    function expandCollapse (btn, collapsed) {
        for(var n = 0; n < piranha.pageedit.blocks.length; n++) {
            piranha.pageedit.blocks[n].meta.isCollapsed = collapsed;
        }
        btn.blur();
    }
</script>
~~~

Then in `Configure` in your `Startup.cs`, add the following:

~~~ csharp
// Custom manager action
Piranha.Manager.Actions.Toolbars.PageEdit.Insert(0, new Piranha.Manager.Extend.ToolbarAction
{
    InternalId = "ExpandCollapse",
    ActionView = "Partial/Actions/_PageExpandCollapse.cshtml"
});
~~~

**Reply by horacioj** [reply]
Perfect!
